crates/noirc_abi/src/input_parser/toml.rs

```diff
@@ -1,32 +1,35 @@
+use super::InputValue;
+use crate::errors::InputParserError;
 use acvm::FieldElement;
 use serde_derive::Deserialize;
 use std::{collections::BTreeMap, path::Path};
 
-use super::InputValue;
-
 pub(crate) fn parse<P: AsRef<Path>>(
     path_to_toml: P,
-) -> Result<BTreeMap<String, InputValue>, String> {
+) -> Result<BTreeMap<String, InputValue>, InputParserError> {
     let path_to_toml = path_to_toml.as_ref();
     if !path_to_toml.exists() {
-        return Err(format!("cannot find input file at located {}, run nargo build to generate the missing Prover and/or Verifier toml files", path_to_toml.display()));
+        return Err(InputParserError::MissingTomlFile(format!("cannot find input file at located {}, run nargo build to generate the missing Prover and/or Verifier toml files", path_to_toml.display())));
```

Comment:
Apologies for all the minor delays in this PR, these error variants should contain the data like path_to_toml directly and the format itself should be delayed until the Display impl is called

Reply:
No worries! I appreciate the notes, as they will help for the future. I just pushed something, and the errors look back as they did before:
<img width="845" alt="Screen Shot 2022-07-20 at 11 13 28 AM" src="https://user-images.githubusercontent.com/43554004/180018565-c04f4848-7ab7-4681-b767-8c695fdf8f74.png">

Let me know if I did it how you were thinking